File: frontend/src/types/analysis.ts
export interface Scores {
  overall: number;
  project_diversity: number;
  language_breadth: number;
  commit_consistency: number;
  readme_quality: number;
  community_engagement: number;
  originality: number;
}

export interface RepoSummary {
  name: string;
  stars: number;
  language: string | null;
  description: string | null;
}

export interface Stats {
  public_repos: number;
  followers: number;
  account_age_days: number;
  total_stars: number;
}

export interface AnalysisResult {
  id: string;
  github_username: string;
  avatar_url: string | null;
  bio: string | null;
  scores: Scores;
  llm_feedback: string | null;
  suggestions: string[] | null;
  top_languages: string[] | null;
  top_repos: RepoSummary[] | null;
  stats: Stats;
  credits_remaining: number;
  created_at: string;
}

export interface HistoryItem {
  id: string;
  github_username: string;
  avatar_url: string | null;
  score_overall: number;
  scores: Scores;
  created_at: string;
}

File: backend/app/schemas/analysis.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class AnalyzeRequest(BaseModel):
    github_username: str = Field(..., min_length=1, max_length=39)


class ScoresResponse(BaseModel):
    overall: float
    project_diversity: float
    language_breadth: float
    commit_consistency: float
    readme_quality: float
    community_engagement: float
    originality: float


class RepoSummary(BaseModel):
    name: str
    stars: int
    language: str | None
    description: str | None


class StatsResponse(BaseModel):
    public_repos: int
    followers: int
    account_age_days: int
    total_stars: int


class AnalysisResponse(BaseModel):
    id: UUID
    github_username: str
    avatar_url: str | None
    bio: str | None
    scores: ScoresResponse
    llm_feedback: str | None
    suggestions: list[str] | None
    top_languages: list[str] | None
    top_repos: list[RepoSummary] | None
    stats: StatsResponse
    credits_remaining: int
    created_at: datetime


class NoCreditsResponse(BaseModel):
    error: str = "no_credits"
    message: str = "You've used all 2 free analyses."
    upgrade_url: str = "/pricing"
